code/API_definitions/EdgeCloud_LcM.yaml

```diff
@@ -0,0 +1,658 @@
+openapi: 3.0.3
+info:
+  title: Edge Cloud API
+  description:  |
+ 
+    Edge Cloud API for artifact managment, application management and resource management.  
+    # Introduction
+    APIs defined in this version of the specification can be categorized into the following areas:
+    * __Application Management__ - Deploy and remove applications within artifacts
+    * __Application Artifacts Management__ - Management of application descriptors, binaries, charts and packages 
+    * __MEC Platform information__ - Retrieves all the Edge nodes available according to some defined parameters
+    # Relevant terms and definitions
+    This section provides definitions of terminologies commonly referred to throughout the API descriptions.
+ 
+    * __Application Provider__ - An application developer, onboarding his/her edge application on operator platform (MEC).    
+    * __Artifact__ - Descriptor, binary image, charts or any other package assosiated with the application.
+    * __OP__ - Operator Platform.
+    * __Region__ - Human readable name of the geografical zone of the MEC. Defined by the OP.
+    # API Functionality
+    __Application Management__
+    * __onboardApplication__ - Submits an application details to an OP. Based on the details provided,  OP shall do bookkeeping, resource validation and other pre-deployment operations. 
+    * __removeApplication__ - Removes an application from an OP
+ 
+    __Application Artifacts Management__
+    * __onboardArtifact__  Submits an artifact to operator platform
+    * __removeArtifact__  Removes an artifact from operator platform
+    __MEC Platform information__
+    *  __retrieveEdgeNodes__ List of the operator’s MECs and their status, ordering the results by location and filtering by status (active/inactive/unknown)
+    # Further info and support
+    (FAQs will be added in a later version of the documentation)
+ 
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: sp-edc@lists.camaraproject.org
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.1.0
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+ 
+servers:
+  - url: "{apiRoot}/{basePath}"
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+      basePath:
+        default: edge-cloud/v1
+        description: Base path for the Edge Cloud API
+ 
+security:
+  - oAuth2ClientCredentials: 
+    - nbi-mgmt
+ 
+tags:
+  - name: Applications
+    description: Manage of Applications
+  - name: Application Artifacts 
+    description: Manage of Artifacts
+  - name: Mobile Edge Cloud Platform
+    description: Mobile Edge Nodes Available
+ 
+paths:
+  /app:
```

Comment:
Shouldnt we keep some AppName or AppId in the path for app provisioning e.g., /app/{appId} instead of keeping AppName in request body?

Reply:
It can be considered once the discussion regarding single app+artifact vs separated is resolved. Our view is that if it finally separated it would be better to keep it this way so to have better sinergy within artifact and application methods.

Reply:
AppName is no longer used in v0.2.0